### PR TITLE
Tsnize more

### PR DESCRIPTION
**Pull request by mizchi**

### CODE REVIEW

File: packages/@githon/markup-it/src/utils/isBlock.ts [hunk 1]
```diff
@@ -1,7 +1,6 @@
-const objectValues = require("object-values");
```
Comment (azu) [inline packages/@githon/markup-it/src/utils/isBlock.ts:1]: Yeah, We can remove `object-values` from `pacakge.json`

File: packages/@githon/markup-it/src/models/rule.ts [hunk 2]
```diff
@@ -9,11 +10,11 @@ const RuleRecord = Immutable.Record({
     type: new String(),
 
     // Listener / Handlers {Map(<String,List<Function>)}
-    listeners: new Immutable.Map(),
+    listeners: Immutable.Map(),
 });
 
 function Rule(type) {
-    if (!(this instanceof Rule)) return new Rule(type);
+    if (!(this instanceof Rule)) return Rule(type);
```
Comment (azu) [inline packages/@githon/markup-it/src/models/rule.ts:17]: It make ∞ loop

Comment (mizchi) [reply packages/@githon/markup-it/src/models/rule.ts:17]: oops

**Comment by azu** [comment]
📝 

> TypeError: Right-hand side of 'instanceof' is not callable

It is cased by `require`  ES module. `require` can not use`.default` property.
`a insteanceof undefined` throw this error.

```diff
diff --git a/packages/@githon/markdown/jest.config.js b/packages/@githon/markdown/jest.config.js
index c7519550..3664125b 100644
--- a/packages/@githon/markdown/jest.config.js
+++ b/packages/@githon/markdown/jest.config.js
@@ -1,6 +1,5 @@
 module.exports = {
     roots: ["src"],
     preset: "ts-jest",
-    setupFilesAfterEnv: ["./testing/setup.js"],
     testEnvironment: "node",
 };
diff --git a/packages/@githon/markdown/package.json b/packages/@githon/markdown/package.json
index a0960795..0579f5d0 100644
--- a/packages/@githon/markdown/package.json
+++ b/packages/@githon/markdown/package.json
@@ -34,7 +34,7 @@
   "scripts": {
     "build": "tsc -p .",
     "clean": "rimraf lib/",
-    "test": "mocha test/"
+    "test": "jest"
   },
   "dependencies": {
     "@githon/html": "^3.3.8",
@@ -42,7 +42,8 @@
     "markdown-escape": "^1.0.1"
   },
   "devDependencies": {
-    "mocha": "^8.0.1"
+    "jest": "^26.0.1",
+    "ts-jest": "^26.1.0"
   },
   "publishConfig": {
     "access": "public"
diff --git a/packages/@githon/markdown/src/__tests__/glossary.js b/packages/@githon/markdown/src/__tests__/glossary.js
index cda8e5b5..9ea4c3e2 100644
--- a/packages/@githon/markdown/src/__tests__/glossary.js
+++ b/packages/@githon/markdown/src/__tests__/glossary.js
@@ -2,7 +2,7 @@ const fs = require("fs");
 const path = require("path");
 const assert = require("assert");
 
-const glossary = require("../../lib").glossary;
+const glossary = require("../").glossary;
 
 describe("Glossary parsing", () => {
     let LEXED;
diff --git a/packages/@githon/markdown/src/__tests__/inline.js b/packages/@githon/markdown/src/__tests__/inline.js
index fa43834f..bd1d7333 100644
--- a/packages/@githon/markdown/src/__tests__/inline.js
+++ b/packages/@githon/markdown/src/__tests__/inline.js
@@ -2,7 +2,7 @@ const fs = require("fs");
 const path = require("path");
 const assert = require("assert");
 
-const inline = require("../../lib").inline;
+const inline = require("../").inline;
 
 describe("Inline", () => {
     it("should render inline markdown", () => {
diff --git a/packages/@githon/markdown/src/__tests__/langs.js b/packages/@githon/markdown/src/__tests__/langs.js
index 41e955cc..42fcf026 100644
--- a/packages/@githon/markdown/src/__tests__/langs.js
+++ b/packages/@githon/markdown/src/__tests__/langs.js
@@ -2,7 +2,7 @@ const fs = require("fs");
 const path = require("path");
 const assert = require("assert");
 
-const langs = require("../../lib").langs;
+const langs = require("../").langs;
 
 describe("Languages parsing", () => {
     let LEXED;
diff --git a/packages/@githon/markdown/src/__tests__/page.js b/packages/@githon/markdown/src/__tests__/page.js
index 4484a6c0..fa8f2798 100644
--- a/packages/@githon/markdown/src/__tests__/page.js
+++ b/packages/@githon/markdown/src/__tests__/page.js
@@ -2,7 +2,7 @@ const fs = require("fs");
 const path = require("path");
 const assert = require("assert");
 
-const page = require("../../lib").page;
+const page = require("../").page;
 
 describe("Page parsing", () => {
     let LEXED;
diff --git a/packages/@githon/markdown/src/__tests__/readme.js b/packages/@githon/markdown/src/__tests__/readme.js
index d457486a..4ac19c1f 100644
--- a/packages/@githon/markdown/src/__tests__/readme.js
+++ b/packages/@githon/markdown/src/__tests__/readme.js
@@ -2,7 +2,7 @@ const fs = require("fs");
 const path = require("path");
 const assert = require("assert");
 
-const readme = require("../../lib").readme;
+const readme = require("../").readme;
 
 describe("Readme parsing", () => {
     let LEXED;
diff --git a/packages/@githon/markdown/src/__tests__/summary.js b/packages/@githon/markdown/src/__tests__/summary.js
index 962fadc1..0d622c20 100644
--- a/packages/@githon/markdown/src/__tests__/summary.js
+++ b/packages/@githon/markdown/src/__tests__/summary.js
@@ -2,7 +2,7 @@ const fs = require("fs");
 const path = require("path");
 const assert = require("assert");
 
-const summary = require("../../lib").summary;
+const summary = require("../").summary;
 
 function lex(fixtureFile) {
     return summary(fs.readFileSync(path.join(__dirname, "fixtures", fixtureFile), "utf8"));
diff --git a/packages/@githon/markdown/src/toHTML.ts b/packages/@githon/markdown/src/toHTML.ts
index 708a8a50..06dff2c8 100644
--- a/packages/@githon/markdown/src/toHTML.ts
+++ b/packages/@githon/markdown/src/toHTML.ts
@@ -12,9 +12,7 @@ const html = new MarkupIt(htmlSyntax);
  * @return {String} (html)
  */
 function convertMdToHTMLBlock(src: string) {
-    // @ts-expect-error
     const content = markdown.toContent(src);
-    // @ts-expect-error
     const textHtml = html.toText(content);
 
     return textHtml;
diff --git a/packages/@githon/markup-it/src/models/content.ts b/packages/@githon/markup-it/src/models/content.ts
index 8ec42e80..ec079347 100644
--- a/packages/@githon/markup-it/src/models/content.ts
+++ b/packages/@githon/markup-it/src/models/content.ts
@@ -1,12 +1,13 @@
 import Immutable from "immutable";
-const Token = require("./token");
-const BLOCKS = require("../constants/blocks");
+import Token from "./token";
+import BLOCKS from "../constants/blocks";
 
 const Content: any = Immutable.Record({
     // Name of the syntax used to parse
     syntax: String(),
 
     // Entry token
+    // @ts-ignore
     token: Token.create(BLOCKS.DOCUMENT),
 });
 
diff --git a/packages/@githon/markup-it/src/models/rule.ts b/packages/@githon/markup-it/src/models/rule.ts
index 913f4dad..db73ac40 100644
--- a/packages/@githon/markup-it/src/models/rule.ts
+++ b/packages/@githon/markup-it/src/models/rule.ts
@@ -1,10 +1,8 @@
 import Immutable from "immutable";
 import is from "is";
+import Token from "./token";
 
 const inherits = require("util").inherits;
-
-const Token = require("./token");
-
 const RuleRecord = Immutable.Record({
     // Type of the rule
     type: new String(),
@@ -13,8 +11,11 @@ const RuleRecord = Immutable.Record({
     listeners: Immutable.Map(),
 });
 
-function Rule(type) {
-    if (!(this instanceof Rule)) return Rule(type);
+function Rule(type?) {
+    if (!(this instanceof Rule)) {
+        // @ts-expect-error
+        return new Rule(type);
+    }
 
     RuleRecord.call(this, {
         type: type,
diff --git a/packages/@githon/markup-it/src/models/rules.ts b/packages/@githon/markup-it/src/models/rules.ts
index d8693b1b..29c25574 100644
--- a/packages/@githon/markup-it/src/models/rules.ts
+++ b/packages/@githon/markup-it/src/models/rules.ts
@@ -1,18 +1,19 @@
 import Immutable from "immutable";
 import is from "is";
-
+import Rule from "./rule";
 const inherits = require("util").inherits;
-
-const Rule = require("./rule");
-
 const RulesSetRecord = Immutable.Record({
     rules: Immutable.List(),
 });
 
 function RulesSet(rules) {
-    if (!(this instanceof RulesSet)) return RulesSet(rules);
+    if (!(this instanceof RulesSet)){
+        // @ts-expect-error workaround instanceof
+        return new RulesSet(rules);
+    }
 
-    rules = RulesList(rules);
+    // @ts-ignore
+    rules = new RulesList(rules);
 
     RulesSetRecord.call(this, {
         rules: rules,
@@ -37,7 +38,8 @@ RulesSet.prototype.add = function (newRules) {
     let rules = this.getRules();
 
     // Prepare new rules
-    newRules = RulesList(newRules);
+    // @ts-ignore
+    newRules = new RulesList(newRules);
 
     // Concat rules
     rules = rules.concat(newRules);
@@ -54,7 +56,8 @@ RulesSet.prototype.unshift = function (newRules) {
     let rules = this.getRules();
 
     // Prepare new rules
-    newRules = RulesList(newRules);
+    // @ts-ignore
+    newRules = new RulesList(newRules);
 
     // Add rules
     rules = rules.unshift.apply(rules, newRules.toArray());
diff --git a/packages/@githon/markup-it/src/models/syntax.ts b/packages/@githon/markup-it/src/models/syntax.ts
index a4cf274b..5e601c92 100644
--- a/packages/@githon/markup-it/src/models/syntax.ts
+++ b/packages/@githon/markup-it/src/models/syntax.ts
@@ -1,26 +1,31 @@
 import Immutable from "immutable";
+import Rule from "./rule";
+import RulesSet from "./rules";
+import defaultRules from "../constants/defaultRules";
 const inherits = require("util").inherits;
 
-const Rule = require("./rule");
-const RulesSet = require("./rules");
-const defaultRules = require("../constants/defaultRules");
-
 const SyntaxSetRecord = Immutable.Record({
     name: String(),
+    // @ts-ignore
     entryRule: new Rule(),
+    // @ts-ignore
     inline: new RulesSet([]),
+    // @ts-ignore
     blocks: new RulesSet([]),
 });
 
 function SyntaxSet(name, def) {
     if (!(this instanceof SyntaxSet)) {
-        return SyntaxSet(name, def);
+        // @ts-ignore
+        return new SyntaxSet(name, def);
     }
 
     SyntaxSetRecord.call(this, {
         name: name,
         entryRule: def.entryRule,
+        // @ts-ignore
         inline: new RulesSet(def.inline),
+        // @ts-ignore
         blocks: new RulesSet(def.blocks),
     });
 }
```
I'Ve passed build, but example/benchmark is missing sidebar

**Comment by azu** [comment]
image

I got it.

**Comment by azu** [comment]
I've merge https://github.com/githon/githon/commit/adb6aa398b0287d544ee7e124f33bffa0f81986f includes this PR.
Thanks.

**Pull request closed without merging**